cpp/src/structs/structs_column_view.cu

```diff
@@ -25,4 +25,20 @@ structs_column_view::structs_column_view(column_view const& rhs) : column_view{r
   CUDF_EXPECTS(type().id() == type_id::STRUCT, "structs_column_view only supports struct columns");
 }
 
+column_view structs_column_view::get_sliced_child(int index) const
+{
+  std::vector<column_view> children;
+  children.reserve(child(index).num_children());
+  for (size_type i = 0; i < child(index).num_children(); i++) {
+    children.push_back(child(index).child(i));
+  }
```

Comment:
This use `cudf::test::make_transform_counting_iterator` and add `const` to the `children` vector as well. I have a PR that is doing this in a few places. I will change it once this PR gets merged.

Reply:
Is it legit to be using cudf::test stuff in production code?

Reply:
I will be moving the `cudf::test::make_transform_counting_iterator` out of test :)